core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigration.scala

```diff
@@ -368,8 +301,131 @@ trait WorkspaceMigrationHistory extends DriverComponent with RawSqlQuery {
+        where #$primaryKey = $key
+      """
+
+    def update3[A, B, C](key: PrimaryKey,
+                         columnName1: ColumnName[A], value1: A,
+                         columnName2: ColumnName[B], value2: B,
+                         columnName3: ColumnName[C], value3: C
+                        )
+                        (implicit
+                         setA: SetParameter[A],
+                         setB: SetParameter[B],
+                         setC: SetParameter[C]
+                        ): ReadWriteAction[Int] =
+      sqlu"""
+        update #$tableName
+        set #$columnName1 = $value1, #$columnName2 = $value2, #$columnName3 = $value3
+        where #$primaryKey = $key
+      """
+
+    def update8[A, B, C, D, E, F, G, H](key: PrimaryKey,
+                                        columnName1: ColumnName[A], value1: A,
+                                        columnName2: ColumnName[B], value2: B,
+                                        columnName3: ColumnName[C], value3: C,
+                                        columnName4: ColumnName[D], value4: D,
+                                        columnName5: ColumnName[E], value5: E,
+                                        columnName6: ColumnName[F], value6: F,
+                                        columnName7: ColumnName[G], value7: G,
+                                        columnName8: ColumnName[H], value8: H
+                                       )
+                                       (implicit
+                                        setA: SetParameter[A],
+                                        setB: SetParameter[B],
+                                        setC: SetParameter[C],
+                                        setD: SetParameter[D],
+                                        setE: SetParameter[E],
+                                        setF: SetParameter[F],
+                                        setG: SetParameter[G],
+                                        setH: SetParameter[H]
+                                       ): ReadWriteAction[Int] =
+      sqlu"""
+        update #$tableName
+        set #$columnName1 = $value1,
+            #$columnName2 = $value2,
+            #$columnName3 = $value3,
+            #$columnName4 = $value4,
+            #$columnName5 = $value5,
+            #$columnName6 = $value6,
+            #$columnName7 = $value7,
+            #$columnName8 = $value8
+        where #$primaryKey = $key
+      """
+
+
+    def upsert[A](key: PrimaryKey, column: ColumnName[A], value: A)
+                 (implicit setA: SetParameter[A])
+    : WriteAction[Int] =
+      sqlu"replace into #$tableName (#$primaryKey, #$column) values ($key, $value)"
+
 
+    def delete: WriteAction[Int] =
+      sqlu"""delete from #$tableName"""
   }
 }
 
-final case class MigrationColumnName(value: String) extends ValueObject
\ No newline at end of file
+final case class ColumnName[-A](value: String) extends ValueObject
```

Comment:
Using a type parameter helps ensure that the type of the column matches the value in `update`